apps/test/unit/gamelab/GameLabSpriteTest.js

```diff
@@ -700,6 +700,174 @@ describe('GameLabSprite', function () {
     });
   });
 
+  describe('animation.goToFrame()', function () {
+    var animation;
+
+    beforeEach(function () {
+      animation = createTestAnimation(10); // with 10 frames
+      animation.frameDelay = 1; // One update() call = one frame
+    });
+
+    it('plays a paused animation when target != current', function () {
+      var start = 2;
+      for (var target = 0; target < 5; target++) {
+        if (target !== start) {
+          animation.changeFrame(start);
+          animation.stop();
+          expect(animation.playing).to.be.false;
+
+          animation.goToFrame(target);
+          expect(animation.playing).to.be.true;
+        }
+      }
+    });
+
+    it('does not play a paused animation when target == current', function () {
+      for (var startAndTarget = 0; startAndTarget < 5; startAndTarget++) {
+        animation.changeFrame(startAndTarget);
+        animation.stop();
+        expect(animation.playing).to.be.false;
+
+        animation.goToFrame(startAndTarget);
+        expect(animation.playing).to.be.false;
+      }
+    });
+
+    it('never pauses a playing animation immediately', function () {
+      var start = 2;
+      for (var target = 0; target < 5; target++) {
+        animation.changeFrame(start);
+        animation.play();
+        expect(animation.playing).to.be.true;
+
+        animation.goToFrame(target);
+        expect(animation.playing).to.be.true;
+      }
+    });
+
+    it('plays the animation forward to the target frame when target > current', function () {
+      animation.changeFrame(1);
+      animation.goToFrame(4);
```

Comment:
you're right this is super confusing and I would've guessed the expected behaviour totally wrong. Thanks for writing tests!